khmer/_oxli/graphs.pyx

```diff
@@ -548,32 +540,35 @@ cdef class Hashgraph(Hashtable):
         cdef shared_ptr[CpHashgraph] this = self._hg_this
 
         with nogil:
-            deref(deref(self._hg_this).partition).find_all_tags(_kmer, deref(tags), 
+            deref(deref(self._hg_this).partition).find_all_tags(_kmer, deref(tags),
                                                                 deref(this).all_tags)
 
         return result
 
-    def consume_seqfile_and_tag(self, str filename):
+    def consume_seqfile_and_tag(self, object parser_or_filename):
         '''Consume all sequences in a FASTA/FASTQ file and tag the resulting
         graph.'''
         cdef unsigned long long n_consumed = 0
         cdef unsigned int total_reads = 0
-        cdef string _filename = _bstring(filename)
+        cdef FastxParserPtr _parser = self._get_parser(parser_or_filename)
+
+        with nogil:
+            deref(self._hg_this).\
+                consume_seqfile_and_tag_readparser[CpFastxReader](_parser,
+                                                                  total_reads,
+                                                                  n_consumed)
```

Comment:
Quick pair programming session with @camillescott revealed I had failed to move these changes over from `feature/cython_cleanup` along with the others. This explained the latest set of failing tests.